### PR TITLE
Clarifying accounts/regions in AWS IRSA setup

### PR DESCRIPTION
**Pull request by ddl-marc-doan**

### CODE REVIEW

File: AWS_SETUP_README.md [hunk 1]
```diff
@@ -104,13 +104,13 @@ Trust Relationship:
...
             "Principal": {
-                "Federated": "arn:aws:iam::{EKS_AWS_ACCOUNT}:oidc-provider/oidc.eks.{EKS_AWS_ACCOUNT_REGION}.amazonaws.com/id/{EKS_OIDC_ID}"
+                "Federated": "arn:aws:iam::{EKS_AWS_ACCOUNT}:oidc-provider/oidc.eks.{EKS_CLUSTER_REGION}.amazonaws.com/id/{EKS_OIDC_ID}"
             },
             "Action": "sts:AssumeRoleWithWebIdentity",
             "Condition": {
                 "StringLike": {
-                    "oidc.eks.{EKS_AWS_ACCOUNT_REGION}.amazonaws.com/id/{EKS_OIDC_ID}:sub": "",
-                    "oidc.eks.{EKS_AWS_ACCOUNT_REGION}.amazonaws.com/id/{EKS_OIDC_ID}:aud": "sts.amazonaws.com"
+                    "oidc.eks.{EKS_AWS_ACCOUNT}.amazonaws.com/id/{EKS_OIDC_ID}:sub": "",
```
Comment (ddl-wadkars) [inline AWS_SETUP_README.md:112]: This is region I believe. Example `https://oidc.eks.us-west-2.amazonaws.com/id/F48A71692D645AEF35A3E1BE605F8BF9`

File: AWS_SETUP_README.md [hunk 2]
```diff
@@ -104,13 +104,13 @@ Trust Relationship:
...
             "Effect": "Allow",
             "Principal": {
-                "Federated": "arn:aws:iam::{EKS_AWS_ACCOUNT}:oidc-provider/oidc.eks.{EKS_AWS_ACCOUNT_REGION}.amazonaws.com/id/{EKS_OIDC_ID}"
+                "Federated": "arn:aws:iam::{EKS_AWS_ACCOUNT}:oidc-provider/oidc.eks.{EKS_CLUSTER_REGION}.amazonaws.com/id/{EKS_OIDC_ID}"
             },
             "Action": "sts:AssumeRoleWithWebIdentity",
             "Condition": {
                 "StringLike": {
-                    "oidc.eks.{EKS_AWS_ACCOUNT_REGION}.amazonaws.com/id/{EKS_OIDC_ID}:sub": "",
-                    "oidc.eks.{EKS_AWS_ACCOUNT_REGION}.amazonaws.com/id/{EKS_OIDC_ID}:aud": "sts.amazonaws.com"
```
Comment (ddl-wadkars) [inline AWS_SETUP_README.md:113]: And we do need this ` "oidc.eks.{EKS_AWS_ACCOUNT_REGION}.amazonaws.com/id/{EKS_OIDC_ID}:aud": "sts.amazonaws.com"`

**Review by ddl-wadkars — CHANGES_REQUESTED** [review]

File: AWS_SETUP_README.md (hunk 1 above)
Comment (ddl-marc-doan) [reply AWS_SETUP_README.md]: Corrected. Thanks!

File: AWS_SETUP_README.md (hunk 2 above)
Comment (ddl-marc-doan) [reply AWS_SETUP_README.md:113]: It's not strictly needed. The default from the [EKS pod identity webhook](https://github.com/aws/amazon-eks-pod-identity-webhook) sets `aud` to `sts.amazonaws.com`. If we were changing it, we'd need to specifically set the new value (both in the trust policy and at the cluster service-account-annotation level).

Comment (ddl-marc-doan) [reply AWS_SETUP_README.md:113]: Removed `aud` lines from trust policy requirements in latest commit.

**Pull request merged**